invenio_app_rdm/records_ui/searchapp.py

```diff
@@ -0,0 +1,230 @@
+    def paginationOptions(self):
+        """Format the pagination options to be used in React-SearchKit JS."""
+        if not getattr(self, 'default_size') or \
+                self.default_size not in self.pagination_options:
+            raise ValueError(
+                'Parameter default_size should be part of pagination_options')
+        return {
+                "resultsPerPage": [
+                    {"text": str(option), "value": option}
+                    for option in self.pagination_options
+                ],
+                "defaultValue": self.default_size,
+            }
+
+    @property
+    def defaultSortingOnEmptyQueryString(self):
+        """Defines the default sorting options when there is no query."""
+        return {
+            "sortBy": self.sort.default_no_query,
+        }
+
+    @classmethod
+    def generate(cls, options, **kwargs):
+        """Create JSON config for React-Searchkit."""
+        generator_object = cls(options)
+        config = {
+            "appId": generator_object.appId,
+            "initialQueryState": generator_object.initialQueryState,
+            "searchApi": generator_object.searchApi,
+            "sortOptions": generator_object.sortOptions,
+            "aggs": generator_object.aggs,
+            "layoutOptions": generator_object.layoutOptions,
+            "sortOrderDisabled": True,
+            "paginationOptions": generator_object.paginationOptions,
+            "defaultSortingOnEmptyQueryString":
+                generator_object.defaultSortingOnEmptyQueryString
+
+        }
+        config.update(kwargs)
+        return config
+
+
+#
+# Application state context processors
+#
+def sort_config(config_name):
+    """Sort configuration."""
+    return SortConfig(
+        current_app.config['RDM_RECORDS_SORT_OPTIONS'],
+        current_app.config[config_name].get('sort', []),
+        current_app.config[config_name].get('sort_default', None),
+        current_app.config[config_name].get('sort_default_no_query'),
+    )
+
+
+def facets_config(config_name):
+    """Facets configuration."""
+    return FacetsConfig(
+        current_app.config['RDM_RECORD_FACETS'],
+        current_app.config[config_name].get('facets', [])
+    )
+
+
+def search_app_config(config_name, endpoint, overrides=None, **kwargs):
+    """Search app config"""
+    opts = dict(
+        endpoint=endpoint,
+        headers = {"Accept": "application/vnd.inveniordm.v1+json"},
+        grid_view=False,
+        sort=sort_config(config_name),
+        facets=facets_config(config_name),
+    )
+    opts.update(kwargs)
+    return SearchAppConfig.generate(opts, **overrides)
+
+
+def search_app_context():
+    """Search app context processor."""
+    return {
+        'search_app_config': partial(
+            search_app_config, '/api/records', 'RDM_RECORD_SEARCH'),
+        'search_drafts_app_config': partial(
+            search_app_config, '/api/user/records', 'RDM_RECORD_SEARCH_DRAFTS')
+    }
```

Comment:
Could those be made to rely on the ServiceConfig? instead of having to extract the configurations again and be wary of variable names? But maybe the layers are too far apart...